realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -665,19 +698,23 @@ public static String getTableName() {
                 }
             }
         }
+        if (json.has("parentObjects")) {
+            throw new io.realm.exceptions.RealmException("\"@LinkingObject\" field \"parentObjects\" cannot be loaded from json");
```

Comment:
Right now we ignore JSON properties not part of the model, so I would prefer this doing the same to be honest. The primary argument being consistency. We can always argue the validity of that choice, but given that I have seen a fair amount of REST API's returning extra stuff like status fields and what not, silently ignoring those fields is actually beneficial.